### DISCUSSION TITLE
Merge my Fork to Master #1158 PR

### DISCUSSION THREAD

**Discussion by JavaJoeS** [eclipse-platform/eclipse.platform#1218] [General]
https://github.com/JavaJoeS/eclipse.platform/tree/master
What are my next steps to get my fork into the master.  According to what Ive read (newbie here) the PR gets merged to my fork and then, if all is well, its merged to master?    Any help, greatly appreciated.   I added a README.md to my core.

**Comment by iloveeclipse** [comment]
Here are steps that would lead to merge on master.

1) Currently your PR has ~60 commits. I'm pretty sure most of them are "work in progress". Squash all commits to reasonable number of changes, like 1 to 3 commits if the different states make sense.
2) Rebase your squashed work on latest master commit
3) Make sure all tests pass and new code has new tests
4) Provide some documentation of the change, because it seem to be a non trivial one
5) Force push to the branch on github to update your PR
6) Invite interesting committers to a review on PR
7) Update PR based on review comments
8) Get a "go" for merge to master

**Reply by JavaJoeS** [reply]
WoW, I really appreciate the recommendation..   
1.  Yes most of my commits are WiP..  Im a newbie, not sure what Squash down commits is or how to accomplish it?  
2.  I see rebase in my eclipse Team pull down..  Correct?
3.  Im a bit lacking in tests, not sure how to set up tests... I copied a few junit tests, but it looked like waste of time., not much gain..
4. Can you look the the README.md that I added to the org.eclipse.core.pki bundle?   Is that sufficient?
5. How do I force push on branch to update the PR? I dont see any instructions on the PR into page..
6. I need all the Interested folks I can get! lol..  Its tuff to do what I have done with these bundles and even tuffer to convey...
7. I know its supposed to be easy to update, but Im button shy after screwing up my bundles, oh about a dozen times...
8. I need that GO, to get rolling...
Thank you for help and for giving me time of day...

**Comment by akurtakov** [comment]
Looking at https://github.com/eclipse-platform/eclipse.platform/pull/1158/files#diff-5a289255643fe69fa92c9c26c71a0bfa03965670d3a1bfde71f29cd85a9d847f this is a lot of code (and even contains a lot of things that don't look related (e.g. ant/*, debug/*, resources/*) to start with.
Requirements to edit eclipse.ini and etc. makes it look like it's not ready to be generally published for everyone.
So may I recommend you different approach?
Build the new bundles on your own, publish them on https://marketplace.eclipse.org/ so people can install and use it even now without having to wait for next Eclipse release.  That should be fully working as you don't seem to touch any other bundle in the repo.
After that if/when you streamline things and there is demand for this feature you can try pushing it to eclipse platform, note that finding someone with time and knowledge on the topic would be a prereq.
